patches/server/0010-Adventure.patch

```diff
@@ -2452,25 +2292,25 @@ index e2def0ca552343143e495736d533b3334686fd62..c87b708c368713a23a10ad97704575ee
                  sender.connection.sendPlayerChatMessage(playerChatMessage, params);
              }
 diff --git a/src/main/java/net/minecraft/network/chat/PlayerChatMessage.java b/src/main/java/net/minecraft/network/chat/PlayerChatMessage.java
-index d89049328641faa889b7c567123ab3a2c63b8df0..81e904e78c9fec49ac244f0a62cef75e5824f2dd 100644
+index d89049328641faa889b7c567123ab3a2c63b8df0..1d5fe82bed18096eb6e2fb1090a52ac92eb4b801 100644
 --- a/src/main/java/net/minecraft/network/chat/PlayerChatMessage.java
 +++ b/src/main/java/net/minecraft/network/chat/PlayerChatMessage.java
 @@ -15,8 +15,54 @@ import net.minecraft.util.SignatureUpdater;
  import net.minecraft.util.SignatureValidator;
  
  public record PlayerChatMessage(
 -    SignedMessageLink link, @Nullable MessageSignature signature, SignedMessageBody signedBody, @Nullable Component unsignedContent, FilterMask filterMask
-+    SignedMessageLink link, @Nullable MessageSignature signature, SignedMessageBody signedBody, @Nullable Component unsignedContent, FilterMask filterMask, @Nullable net.minecraft.network.chat.ChatDecorator.Result result // Paper - adventure; support signed messages
++    SignedMessageLink link, @Nullable MessageSignature signature, SignedMessageBody signedBody, @Nullable Component unsignedContent, FilterMask filterMask, @Nullable net.kyori.adventure.text.Component result // Paper - adventure; support signed messages
  ) {
 +    // Paper start - adventure; support signed messages
 +    public PlayerChatMessage(SignedMessageLink link, @Nullable MessageSignature signature, SignedMessageBody signedBody, @Nullable Component unsignedContent, FilterMask filterMask) {
 +        this(link, signature, signedBody, unsignedContent, filterMask, null);
 +    }
-+    public PlayerChatMessage withResult(net.minecraft.network.chat.ChatDecorator.Result result) {
-+        final PlayerChatMessage msg = this.withUnsignedContent(result.component());
-+        return new PlayerChatMessage(msg.link, msg.signature, msg.signedBody, msg.unsignedContent, msg.filterMask, result);
++    public PlayerChatMessage withResult(Component result) {
++        final PlayerChatMessage msg = this.withUnsignedContent(result);
++        return new PlayerChatMessage(msg.link, msg.signature, msg.signedBody, msg.unsignedContent, msg.filterMask, io.papermc.paper.adventure.PaperAdventure.asAdventure(result));
 +    }
-+    public net.minecraft.network.chat.ChatDecorator.Result requireResult() {
++    public net.kyori.adventure.text.Component requireResult() {
 +        return Objects.requireNonNull(this.result, "Requires a decoration result to be set here");
```

Comment:
I think we can get rid of the extra parameter all together. The result of the decoration should just be set as the unsigned content of the PlayerChatMessage like vanilla does it.